tests/basic/volume_ops_test.go

```diff
@@ -1484,3 +1488,135 @@ var _ = Describe("{ResizeVolumeAfterFull}", func() {
+			"Create fast path volume, reboot the node, check fastpath is active", nil, testrailID)
+		runID = testrailuttils.AddRunsToMilestone(testrailID)
+		log.Infof("The runID  %v ", runID)
+	})
+
+	var pxNode node.Node
+	var contexts []*scheduler.Context
+	var applist = Inst().AppList
+	stepLog := "Create fastpath Volume Reboot Node and check if fastpath is active"
+	It(stepLog, func() {
+		log.InfoD(stepLog)
+		stepLog = "Step 1: Get all the Storage nodes and select a node for test"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			// Get all the Nodes
+			pxNodes, err := GetStorageNodes()
+			log.FailOnError(err, "Unable to get the storage nodes")
+
+			// Select random Storage node for the test
+			if len(pxNodes) > 0 {
+				pxNode = GetRandomNode(pxNodes)
+			} else {
+				log.FailOnError(errors.New("No Storage Node Availiable"), "Error occured while selecting StorageNode")
+			}
+
+			log.Infof("The Selected node for Fast path label is %v : ", pxNode.Name)
+
+			// Remove if node-type label is set before the test
+			Inst().AppList = applist
+			RemoveLabelsAllNodes(k8s.NodeType, true, false)
+		})
+
+		stepLog = "Step 2: Schedule application and Add label on the selected storage Node"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			var err error
+
+			// Add label on the selected node
+			err = Inst().S.AddLabelOnNode(pxNode, k8s.NodeType, k8s.FastpathNodeType)
+			log.FailOnError(err, fmt.Sprintf("Failed add label on node %s", pxNode.Name))
+			Inst().AppList = []string{"fio-fastpath-repl1"}
+			contexts = make([]*scheduler.Context, 0)
+			for i := 0; i < Inst().GlobalScaleFactor; i++ {
+				contexts = append(contexts, ScheduleApplications(fmt.Sprintf("fastpath-%d", i))...)
+			}
+			ValidateApplications(contexts)
+		})
+
+		stepLog = " Step 3: Get app volumes and Check fast path is active on the node"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			var err error
+			for _, ctx := range contexts {
+				var appVolumes []*volume.Volume
+				stepLog = fmt.Sprintf("get volumes for %s app", ctx.App.Key)
+				Step(stepLog, func() {
+					log.InfoD(stepLog)
+					appVolumes, err = Inst().S.GetVolumes(ctx)
+
+					// Get the app volume details
+					log.FailOnError(err, "Failed to get volumes for app %s", ctx.App.Key)
+					dash.VerifyFatal(len(appVolumes) > 0, true, "App volumes exist?")
+					log.Infof("App volumes details: %v ", appVolumes)
+
+					// Loop through the apps and check if the volumes are fastpath active before reboot
+					for appvolume := range appVolumes {
+						log.Infof("current volume : %v", appvolume)
+						if strings.Contains(ctx.App.Key, fastpathAppName) {
+							err := ValidateFastpathVolume(ctx, opsapi.FastpathStatus_FASTPATH_ACTIVE)
+							log.FailOnError(err, "fastpath volume validation failed")
+						}
+					}
+
+				})
+			}
+		})
+		stepLog = " Step 4: Reboot the node wait for it to complete"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			rebootNodeAndWaitForReady(&pxNode)
```

Comment:
before rebooting can you make sure the volume is attached to this node only

Reply:
Fixed. The new check via IsVolumeAttachedOnNode will take care of this case